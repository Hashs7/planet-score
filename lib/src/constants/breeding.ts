/**
 *
 * @type {{MEDIUM: string, BAD: string, GOOD: string}}
 */
export enum BREEDING_CONDITION {
  GOOD = 'v',
  MEDIUM = 'o',
  BAD = 'r'
}

/**
 * Color state for breeding score
 *
 * @type {{[p: string]: string}}
 */
export const BREEDING_CONDITION_COLOR: { [index: string]: string } = {
  [BREEDING_CONDITION.GOOD]: '#007C3B',
  [BREEDING_CONDITION.MEDIUM]: '#F07E02',
  [BREEDING_CONDITION.BAD]: '#E33116'
};

/**
 * List of available animals
 *
 * @type {{OTHER: string, BEE: string, FARMING_FISH: string, FISHING_FISH: string, CHICKEN: string, SHEEP: string, DUCK: string, COW: string, GOAT: string, RABBIT: string, PIG: string}}
 */
export enum BREEDING_ANIMAL {
  COW = 'bovin',
  CHICKEN = 'poule',
  SHEEP = 'brebis',
  GOAT = 'chevre',
  PIG = 'porc'

  // Icon not integrated
  // BEE = 'abeille',
  // DUCK = 'canard',
  // RABBIT = 'lapin',
  // FARMING_FISH = 'poisson_aquaculture',
  // FISHING_FISH = 'poisson_peche',
  // OTHER = 'autre'
}

/**
 * Regex to extract only breeding animal word
 *
 * @type {RegExp}
 */
export const BREEDING_ANIMAL_REGEX = new RegExp(`(${Object.values(BREEDING_ANIMAL).join('|')})`);
