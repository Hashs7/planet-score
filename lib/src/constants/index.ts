export * from './breeding';
export * from './score';
