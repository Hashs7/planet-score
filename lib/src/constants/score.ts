/**
 * The score values.
 */
export enum ScoreValues {
  A = 'A',
  B = 'B',
  C = 'C',
  D = 'D',
  E = 'E'
}

/**
 * The score values in order of best to worst.
 */
export const SCORE_POSITIONS = [
  ScoreValues.A,
  ScoreValues.B,
  ScoreValues.C,
  ScoreValues.D,
  ScoreValues.E
];
