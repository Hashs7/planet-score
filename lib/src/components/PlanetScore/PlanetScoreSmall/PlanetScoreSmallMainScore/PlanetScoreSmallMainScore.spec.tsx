import { render } from '@testing-library/react';
import PlanetScoreSmallMainScore from './PlanetScoreSmallMainScore';
import { SCORE_POSITIONS } from '@/constants';
import { SvgWrapper } from '@tests/utils';

describe('<PlanetScoreSmallMainScore>', () => {
  SCORE_POSITIONS.forEach((score) => {
    it(`(${score}) renders the PlanetScoreSmallMainScore`, () => {
      const { container } = render(<PlanetScoreSmallMainScore score={score} />, {
        wrapper: SvgWrapper
      });

      expect(container).toMatchSnapshot();
    });
  });
});
