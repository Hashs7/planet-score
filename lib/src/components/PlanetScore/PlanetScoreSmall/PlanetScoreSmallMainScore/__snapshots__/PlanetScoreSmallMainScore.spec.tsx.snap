// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanetScoreSmallMainScore> (A) renders the PlanetScoreSmallMainScore 1`] = `
<div>
  <svg>
    <g
      id="score-A"
    >
      <ellipse
        cx="235"
        cy="458.7"
        rx="117"
        ry="117"
        style="fill: #027D3D;"
        transform="matrix(0.7071 -0.7071 0.7071 0.7071 -255.564 300.5128)"
      />
      <path
        d="M324.3,369.4c-23.9-23.9-55.6-37-89.4-37c-33.8,0-65.5,13.1-89.4,37c-23.9,23.9-37,55.6-37,89.4
			s13.1,65.5,37,89.4s55.6,37,89.4,37c33.8,0,65.5-13.1,89.4-37c23.9-23.9,37-55.6,37-89.4S348.2,393.3,324.3,369.4z M118,458.7
			c0-64.6,52.4-117,117-117s117,52.4,117,117s-52.4,117-117,117S118,523.4,118,458.7z"
        style="fill: #FFFFFF;"
      />
      <path
        d="M169,512.5l43.4-117.3c2-5.4,5.1-9.7,9.1-13c4-3.3,8.7-4.9,13.9-4.9c5.5,0,10.3,1.6,14.3,4.8
			c4,3.2,6.9,7.6,8.8,13.1l42.3,117.3c1.1,3.1,1.7,5.8,1.7,8c0,2.7-0.7,5.2-2.1,7.4c-1.4,2.2-3.2,3.9-5.4,5.1
			c-2.2,1.2-4.7,1.8-7.4,1.8c-8,0-13.3-3.7-15.8-11.1l-6.4-20.2h-59.9l-6.5,20c-2.4,7.6-7.5,11.3-15.5,11.3c-4.4,0-8.2-1.5-11.3-4.4
			c-3.1-2.9-4.7-6.5-4.7-10.7C167.5,517.6,168,515.2,169,512.5z M235.2,411l-20.4,62.5h40.9L235.6,411H235.2z"
        style="fill: #FFFFFF;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreSmallMainScore> (B) renders the PlanetScoreSmallMainScore 1`] = `
<div>
  <svg>
    <g
      id="score-B"
    >
      <ellipse
        cx="235"
        cy="458.7"
        rx="117"
        ry="117"
        style="fill: #7AB72A;"
        transform="matrix(0.7071 -0.7071 0.7071 0.7071 -255.564 300.5128)"
      />
      <path
        d="M324.3,369.4c-23.9-23.9-55.6-37-89.4-37c-33.8,0-65.5,13.1-89.4,37c-23.9,23.9-37,55.6-37,89.4
			s13.1,65.5,37,89.4s55.6,37,89.4,37c33.8,0,65.5-13.1,89.4-37c23.9-23.9,37-55.6,37-89.4S348.2,393.3,324.3,369.4z M118,458.7
			c0-64.6,52.4-117,117-117s117,52.4,117,117s-52.4,117-117,117S118,523.4,118,458.7z"
        style="fill: #FFFFFF;"
      />
      <path
        d="M183.1,517.1V397c0-5,1.6-8.9,4.8-11.8c3.2-2.8,7.2-4.2,12.2-4.2h41.1c7.4,0,14.2,1.6,20.3,4.7
			c6.1,3.1,11,7.6,14.6,13.3c3.6,5.7,5.4,12.3,5.4,19.6c0,14.8-6.6,25.3-19.9,31.5v0.4c4.5,0.6,8.7,2,12.7,4.1
			c4,2.1,7.5,4.8,10.5,8.1s5.3,6.9,7,11c1.6,4,2.4,8.5,2.4,13.3c0,14.8-4.5,26.2-13.5,34.3c-9,8.1-21.1,12.1-36.1,12.1h-44.1
			c-5,0-9.1-1.5-12.3-4.5C184.7,525.9,183.1,522,183.1,517.1z M215.2,409.2v32.1h16.7c5.3,0,9.5-1.5,12.7-4.4
			c3.1-3,4.7-7.1,4.7-12.4c0-4.9-1.6-8.6-4.9-11.3c-3.2-2.6-7.4-4-12.5-4H215.2z M215.2,469.5v35.7h25.3c6,0,10.9-1.7,14.6-5
			c3.7-3.3,5.6-7.9,5.6-13.8c0-5.2-1.9-9.3-5.8-12.3c-3.9-3-9.5-4.6-17-4.6H215.2z"
        style="fill: #FFFFFF;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreSmallMainScore> (C) renders the PlanetScoreSmallMainScore 1`] = `
<div>
  <svg>
    <g
      id="score-C"
    >
      <ellipse
        cx="235"
        cy="458.7"
        rx="117"
        ry="117"
        style="fill: #FBC404;"
        transform="matrix(0.7071 -0.7071 0.7071 0.7071 -255.564 300.5128)"
      />
      <path
        d="M324.3,369.4c-23.9-23.9-55.6-37-89.4-37c-33.8,0-65.5,13.1-89.4,37c-23.9,23.9-37,55.6-37,89.4
			s13.1,65.5,37,89.4s55.6,37,89.4,37c33.8,0,65.5-13.1,89.4-37c23.9-23.9,37-55.6,37-89.4S348.2,393.3,324.3,369.4z M118,458.7
			c0-64.6,52.4-117,117-117s117,52.4,117,117s-52.4,117-117,117S118,523.4,118,458.7z"
        style="fill: #FFFFFF;"
      />
      <path
        d="M167,456.7c0-11.2,1.8-21.6,5.4-31.3c3.6-9.7,8.7-18.1,15.3-25.4c6.6-7.2,14.5-12.8,23.5-16.8
			c9.1-4,19-5.9,29.9-5.9c4.4,0,9.4,0.4,14.8,1.3c5.5,0.8,10.5,2.2,15.2,3.9c4.7,1.8,8.6,4.3,11.7,7.4c3.1,3.2,4.7,7,4.7,11.6
			c0,3.9-1.3,7.3-3.8,10.2s-5.7,4.3-9.5,4.3c-2.3,0-5.6-0.8-9.7-2.4c-4.2-1.6-7.8-2.9-10.8-3.8c-3.1-0.9-7.2-1.4-12.5-1.4
			c-6.6,0-12.4,1.3-17.5,3.8c-5.1,2.6-9.3,6.1-12.6,10.5c-3.4,4.4-5.9,9.6-7.7,15.6c-1.8,6-2.7,12.3-2.7,19.1
			c0,13.5,3.6,24.9,10.7,34c7.2,9.1,17.1,13.7,29.8,13.7c5.3,0,9.6-0.5,12.7-1.6c3.1-1.1,6.9-2.6,11.4-4.5c4.5-2,7.9-2.9,10.3-2.9
			c4.4,0,7.9,1.7,10.4,5c2.5,3.3,3.8,6.5,3.8,9.6c0,6.3-2.7,11.4-8.1,15.2c-5.4,3.8-11.8,6.5-19.3,8c-7.5,1.5-14.5,2.3-21.1,2.3
			c-14.3,0-27.1-3.5-38.4-10.5c-11.3-7-20-16.5-26.3-28.7C170.1,484.8,167,471.4,167,456.7z"
        style="fill: #FFFFFF;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreSmallMainScore> (D) renders the PlanetScoreSmallMainScore 1`] = `
<div>
  <svg>
    <g
      id="score-D"
    >
      <ellipse
        cx="235"
        cy="458.7"
        rx="117"
        ry="117"
        style="fill: #ED7A06;"
        transform="matrix(0.7071 -0.7071 0.7071 0.7071 -255.564 300.5128)"
      />
      <path
        d="M324.3,369.4c-23.9-23.9-55.6-37-89.4-37c-33.8,0-65.5,13.1-89.4,37c-23.9,23.9-37,55.6-37,89.4
			s13.1,65.5,37,89.4s55.6,37,89.4,37c33.8,0,65.5-13.1,89.4-37c23.9-23.9,37-55.6,37-89.4S348.2,393.3,324.3,369.4z M118,458.7
			c0-64.6,52.4-117,117-117s117,52.4,117,117s-52.4,117-117,117S118,523.4,118,458.7z"
        style="fill: #FFFFFF;"
      />
      <path
        d="M175.8,517.2V396.6c0-4.9,1.6-8.9,4.7-12c3.2-3.1,7.1-4.6,11.9-4.6h35.8c12.1,0,22.8,1.9,32.3,5.6
			c9.4,3.7,17.4,9,23.9,15.8c6.5,6.8,11.4,15.1,14.7,24.8c3.3,9.7,5,20.6,5,32.6c0,11.2-1.7,21.4-5.2,30.6
			c-3.5,9.2-8.5,17.1-15,23.6c-6.5,6.6-14.4,11.6-23.6,15.1c-9.2,3.5-19.2,5.3-30,5.3h-37.5c-2.7,0-5.3-0.5-7.8-1.6
			c-2.6-1.1-4.7-2.8-6.5-5.2C176.7,524.2,175.8,521.1,175.8,517.2z M208.1,409.7v93.9h21.8c9.1,0,16.6-2,22.7-6
			c6-4,10.5-9.5,13.5-16.4c3-6.9,4.4-14.8,4.4-23.8c0-9.3-1.5-17.5-4.5-24.7c-3-7.2-7.7-12.8-14.1-16.9c-6.4-4-14.4-6-24-6H208.1z"
        style="fill: #FFFFFF;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreSmallMainScore> (E) renders the PlanetScoreSmallMainScore 1`] = `
<div>
  <svg>
    <g
      id="score-E"
    >
      <ellipse
        cx="235"
        cy="458.7"
        rx="117"
        ry="117"
        style="fill: #E33116;"
        transform="matrix(0.7071 -0.7071 0.7071 0.7071 -255.564 300.5128)"
      />
      <path
        d="M324.3,369.4c-23.9-23.9-55.6-37-89.4-37c-33.8,0-65.5,13.1-89.4,37c-23.9,23.9-37,55.6-37,89.4
			s13.1,65.5,37,89.4s55.6,37,89.4,37c33.8,0,65.5-13.1,89.4-37c23.9-23.9,37-55.6,37-89.4S348.2,393.3,324.3,369.4z M118,458.7
			c0-64.6,52.4-117,117-117s117,52.4,117,117s-52.4,117-117,117S118,523.4,118,458.7z"
        style="fill: #FFFFFF;"
      />
      <path
        d="M194.3,517.2V398.7c0-5.6,1.4-10.1,4.3-13.6c2.9-3.4,7.1-5.2,12.5-5.2h53.5c4.9,0,8.7,1.5,11.3,4.4
			c2.6,2.9,3.9,6.4,3.9,10.5c0,2.7-0.6,5.1-1.8,7.3c-1.2,2.2-2.9,4-5.2,5.4c-2.3,1.4-5,2.1-8.3,2.1h-38v31.4h35.4
			c5.1,0,8.9,1.4,11.7,4.3c2.7,2.9,4.1,6.4,4.1,10.6c0,3.9-1.4,7.4-4.2,10.4c-2.8,3-6.7,4.5-11.6,4.5h-35.4v32.7h39.9
			c5,0,8.8,1.5,11.4,4.4c2.6,2.9,3.9,6.4,3.9,10.6c0,2.3-0.6,4.7-1.8,7.1c-1.2,2.4-2.9,4.3-5.3,5.7c-2.3,1.4-5.1,2.1-8.3,2.1h-55.6
			c-4.8,0-8.7-1.5-11.9-4.5C195.8,526,194.3,522.1,194.3,517.2z"
        style="fill: #FFFFFF;"
      />
    </g>
  </svg>
</div>
`;
